app/pb_kits/playbook/pb_gauge/_gauge.jsx

```diff
@@ -0,0 +1,105 @@
+/* @flow */
+
+import React, { useEffect, useRef } from 'react'
+import { pbChart } from '../'
+import { spacing } from '../utilities/spacing.js'
+import classnames from 'classnames'
+import Highcharts from 'highcharts'
+
+import {
+  buildAriaProps,
+  buildCss,
+  buildDataProps,
+} from '../utilities/props'
+
+type GaugeProps = {
+  aria: Object,
+  className?: String,
+  chartData?: Array,
+  data?: Object,
+  fullCircle: Boolean,
+  height: String,
+  id?: String,
+  max: Number,
+  min: Number,
+  prefix: String,
+  showLabels: Boolean,
+  style: String,
+  suffix: String,
+  title: String,
+  tooltipHtml: String,
+}
+
+const Gauge = (props: GaugeProps) => {
+  const {
+    aria = {},
+    className,
+    chartData = [{ name: 'Name', value: 0 }],
+    data = {},
+    fullCircle = false,
+    height = null,
+    id,
+    max = 100,
+    min = 0,
+    prefix = '',
+    showLabels = false,
+    style = 'solidgauge',
+    suffix = '',
+    title = '',
+    tooltipHtml = '<span style="font-weight: bold; color:{point.color};">●</span>{point.name}: ' + '<b>{point.y}</b>',
+  } = props
+
+  const ariaProps = buildAriaProps(aria)
+  const dataProps = buildDataProps(data)
+
+  const css = buildCss({
+    'pb_gauge_kit': true,
+  })
+  // Runs first time component Renders
+  useEffect(() => {
+    chartData.forEach((obj) => {
+      obj.y = obj.value
+      delete obj.value
+    })
+
+    new pbChart('.selector', {
+      id: id,
+      chartData: chartData,
+      circumference: fullCircle ? [0, 360] : [-100, 100],
+      height: height,
+      min: min,
+      max: max,
+      prefix: prefix,
+      title: title,
+      suffix: suffix,
+      showLabels: showLabels,
+      style: style,
+      tooltipHtml: tooltipHtml,
+      type: 'gauge',
+    })
+  }, [])
+
+  const componentDidMount = useRef(false)
+  // Doesn't run the first time but runs every subsequent render
+  useEffect(() => {
```

Comment:
Is it the intent to have this second `useEffect` executed on **every** subsequent mount or only if the chart data is changed?

Reply:
Only if chartData prop is changed, is it possible to scope rerenders to a specific prop?  Maybe by adding it to that second array argument in `useEffect()`?

Reply:
That's correct 👍🏼 you can scope rerenders to specific props by passing a dependencies array as the second argument of useEffect with the props. That will tell React to only trigger the effect block when any of the props in that array have their reference updated. [Here is a pretty insightful resource](https://overreacted.io/a-complete-guide-to-useeffect/) on different React hooks topics in case you feel like digging a little deeper in the subject 😁